consul-lambda-registrator/trigger.go

```diff
@@ -6,19 +6,21 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/aws/aws-sdk-go-v2/aws"
-	"github.com/aws/aws-sdk-go-v2/service/lambda"
-
 	"github.com/hashicorp/consul/api"
 	"github.com/hashicorp/go-multierror"
+
+	"github.com/hashicorp/terraform-aws-consul-lambda-registrator/consul-lambda-registrator/client"
+	"github.com/hashicorp/terraform-aws-consul-lambda-registrator/consul-lambda-registrator/structs"
 )
 
 const (
 	prefix                = "serverless.consul.hashicorp.com/v1alpha1/lambda"
 	enabledTag            = prefix + "/enabled"
 	arnTag                = prefix + "/arn"
-	payloadPassthroughTag = prefix + "/payload-passhthrough"
+	payloadPassthroughTag = prefix + "/payload-passthrough"
 	regionTag             = prefix + "/region"
+	serviceNameTag        = prefix + "/service-name"
```

Comment:
How do we manage uniqueness when users can specify service names? I'm worried that this enables users to squash other people's services.

Reply:
That's a really good point. I was thinking we would want to provide this as a UX for users to configure their Lambda service names to be different from the function name if they want to. Similar to how we allow this for the ECS integration. But I think you're correct that we shouldn't allow this in the lambda case because lambda already enforces unique function names.